### ISSUE TITLE
Improve error handling in Server:prepareForRecord

### ISSUE THREAD

**Issue by bagong** [supercollider/supercollider#1580]
Windows 8.1, SC alpha1, VS build, 64 bit
Files via
s.prepareForRecord("C:\\Users\\name\\Documents\\SuperCollider\\test\\test.wav")
or
s.prepareForRecord("C:/Users/name/Documents/SuperCollider/test/test.wav")

can only be created, if all folders in a path already exist on the system. In any other case file-creation fails.

**Comment by jamshark70** [comment]
Why is this even a bug? I think it would be rather extraordinary to expect the server, or even the language, to create directories automatically.

Sorry, but I think this should just be closed.

**Comment by bagong** [comment]
Really, well you obviously have some technical insight, I am missing. I naively assume it should be possible to create any path you can write and you have write access to. If it's really absurd, please close. I am on the road. 
However it is certainly counter intuitive and sc shouldn't tacitly write just nothing in that case.

**Comment by jamshark70** [comment]
What I mean is: Standard behavior, in any programming language, when you try to open a file for writing and the location doesn't exist, is to fail to open the file. One rationale is that you wouldn't want typos to create spurious directories in the file system.

In Linux, scsynth posts an error in this case that the path couldn't be opened. To me, this is the expected behavior. If the error doesn't appear in Windows, that would be a bug.

**Comment by bagong** [comment]
Well, you get an error, when you run prepareForRecord as an individual command. But in a series of commands the message won't turn up and recording just writes nothing. But it's certainly not the type of bug I believed it to be...

**Comment by bagong** [comment]
Can you remove the tags for me. I am on a mobile.

**Comment by jamshark70** [comment]
I'll mark this as an enhancement for future (not 3.7). It's a legitimate problem that the language gives no sort of result to the user that the file-open failed. It can do so (see http://doc.sccode.org/Reference/Server-Command-Reference.html#/fail):

```
o = OSCFunc({ |msg| msg.postln }, '/fail', s.addr);
b = Buffer.alloc(s, 65536, 1);
b.write("~/doesntexist/nope.aiff".standardizePath, leaveOpen: true);
--> posts:
File '/home/dlm/doesntexist/nope.aiff' could not be opened: System error : No such file or directory.
[ /fail, /b_write, File '/home/dlm/doesntexist/nope.aiff' could not be opened: System error : No such file or directory.
, 0 ]

b.close;
b.free;
```

`prepareForRecord` could detect this and do something... throw an error that the user could catch? (Tricky b/c this is asynchronous.) Also, /b_write sends /done back to the client if successful, so we could use that to determine success and clean up OSCFuncs.

**Comment by jamshark70** [comment]
Shouldn't be done until #1581 is fixed.

**Comment by telephon** [comment]
I'll add it in #1547, which will go into 3.8

**Comment by bagong** [comment]
Addressed in #1655
